src/tlo/methods/cervical_cancer.py

```diff
@@ -0,0 +1,1515 @@
+                        or df.at[person_id, 'ce_hpv_cc_status'] == 'stage2b'
+                        or df.at[person_id, 'ce_hpv_cc_status'] == 'stage3'
+                        or df.at[person_id, 'ce_hpv_cc_status'] == 'stage4'):
+            hs.schedule_hsi_event(
+                hsi_event=HSI_CervicalCancer_Biopsy(
+                    module=self.module,
+                    person_id=person_id
+                ),
+                priority=0,
+                topen=self.sim.date,
+                tclose=None
+            )
+
+        # sy_chosen_via_screening_for_cin_cervical_cancer reset to 0
+        if df.at[person_id, 'sy_chosen_xpert_screening_for_hpv_cervical_cancer'] == 2:
+            self.sim.modules['SymptomManager'].change_symptom(
+                person_id=person_id,
+                symptom_string='chosen_xpert_screening_for_hpv_cervical_cancer',
+                add_or_remove='-',
+                disease_module=self.module
+                )
+
+        df.at[person_id, 'ce_selected_for_xpert_this_month'] = False
+
+
+
+class HSI_CervicalCancerPresentationVaginalBleeding(HSI_Event, IndividualScopeEventMixin):
+
+    def __init__(self, module, person_id):
+        super().__init__(module, person_id=person_id)
+
+        self.TREATMENT_ID = "CervicalCancer_presentation_vaginal_bleeding"
+        self.EXPECTED_APPT_FOOTPRINT = self.make_appt_footprint({"Over5OPD": 1})
+        self.ACCEPTED_FACILITY_LEVEL = '1a'
+
+    def apply(self, person_id, squeeze_factor):
+        df = self.sim.population.props
+        person = df.loc[person_id]
+        hs = self.sim.modules["HealthSystem"]
+        p = self.sim.modules['CervicalCancer'].parameters
+
+        # Ignore this event if the person is no longer alive:
+        if not person.is_alive:
+            return hs.get_blank_appt_footprint()
+
+        random_value = random.random()
+
+        if random_value <= p['prob_referral_biopsy_given_vaginal_bleeding']:
+            hs.schedule_hsi_event(
+                hsi_event=HSI_CervicalCancer_Biopsy(
+                    module=self.module,
+                    person_id=person_id
+                ),
+                priority=0,
+                topen=self.sim.date,
+                tclose=None
+            )
+
+class HSI_CervicalCancer_Biopsy(HSI_Event, IndividualScopeEventMixin):
+
+    def __init__(self, module, person_id):
+        super().__init__(module, person_id=person_id)
+
+#       print(person_id, self.sim.date, 'vaginal_bleeding_hsi_called -1')
+
+        self.TREATMENT_ID = "CervicalCancer_Biopsy"
+
+        self.EXPECTED_APPT_FOOTPRINT = self.make_appt_footprint({"Over5OPD": 1})
+        self.ACCEPTED_FACILITY_LEVEL = '3'
+
+    def apply(self, person_id, squeeze_factor):
+        df = self.sim.population.props
+        hs = self.sim.modules["HealthSystem"]
+
+        # Ignore this event if the person is no longer alive:
+        if not df.at[person_id, 'is_alive']:
+            return hs.get_blank_appt_footprint()
```

Comment:
@tbhallett is this not being handled already by the Healthsystem? If yes then I think we will save some computational time by removing it here and in all other HSI's.

Reply:
Thanks - now removed from all HSIs